dev-libs/libsmb2/libsmb2-3.0.0.recipe

```diff
@@ -0,0 +1,80 @@
+SUMMARY="SMB2/3 userspace client"
+DESCRIPTION="Libsmb2 is a userspace client library for accessing SMB2/SMB3 \
+shares on a network.
+It is high performance and fully async. It supports both zero-copy \
+for SMB READ/WRITE commands as well as compounded commands."
+HOMEPAGE="https://github.com/sahlberg/libsmb2"
+COPYRIGHT="2020 Ronnie Sahlberg"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://github.com/sahlberg/libsmb2/archive/refs/tags/v${portVersion}.tar.gz"
+CHECKSUM_SHA256="36d10d1628ad34d56646bf9e7005a8abe8ce6b67ba3a949d537ae3e5f08f701f"
+PATCHES="libsmb2-$portVersion.patchset"
+
+ARCHITECTURES="all"
+SECONDARY_ARCHITECTURES="?x86_gcc2 ?x86"
+
+libVersion="3.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	libsmb2$secondaryArchSuffix = $portVersion
+	lib:libsmb2$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libsmb2${secondaryArchSuffix}_devel = $portVersion
+	devel:libsmb2$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	libsmb2$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libssl$secondaryArchSuffix
```

Comment:
Will libssl really used?

Reply:
Not sure if it's used but configure checks for it (if I disable it I get, and even if it's disabled):
```
-- Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY OPENSSL_INCLUDE_DIR) 
```

Reply:
asking because libssl doesn't come in REQUIRES.